internal/util/update/resource-merge.go

```diff
@@ -110,3 +119,130 @@ func (u ResourceMergeUpdater) updatedKptfile(updatedPath string, options UpdateO
 	err = kf.MergeOpenAPI(options.KptFile)
 	return kf, err
 }
+
+// replaceNonKRMFiles replaces the non KRM files in localDir with the corresponding files in updatedDir,
+// it also deletes non KRM files and sub dirs which are present in localDir and not in updatedDir
+func ReplaceNonKRMFiles(updatedDir, originalDir, localDir string) error {
+	updatedSubDirs, updatedFiles, err := getSubDirsAndNonKrmFiles(updatedDir)
+	if err != nil {
+		return err
+	}
+
+	originalSubDirs, originalFiles, err := getSubDirsAndNonKrmFiles(originalDir)
+	if err != nil {
+		return err
+	}
+
+	localSubDirs, localFiles, err := getSubDirsAndNonKrmFiles(localDir)
+	if err != nil {
+		return err
+	}
+
+	// identify all non KRM files modified locally, to leave them untouched
+	locallyModifiedFiles := sets.String{}
+	for _, file := range localFiles.List() {
+		if !originalFiles.Has(file) {
+			// new local file has been added
+			locallyModifiedFiles.Insert(file)
+			continue
+		}
+		same, err := compareFiles(filepath.Join(originalDir, file), filepath.Join(localDir, file))
+		if err != nil {
+			return err
+		}
+		if !same {
+			// local file has been modified
+			locallyModifiedFiles.Insert(file)
+			continue
+		}
+
+		// remove the file from local if it is not modified and is deleted from updated upstream
+		if !updatedFiles.Has(file) {
+			if err = os.Remove(filepath.Join(localDir, file)); err != nil {
+				return err
+			}
+		}
+	}
+
+	// make sure local has all sub-dirs present in updated
+	for _, dir := range updatedSubDirs.List() {
+		if err = os.MkdirAll(filepath.Join(localDir, dir), 0700); err != nil {
```

Comment:
Consider taking the file permissions of the updated dir rather than using 0700

Reply:
https://github.com/kubernetes-sigs/kustomize/blob/master/kyaml/kio/pkgio_writer.go#L77 

We are dirs with default 0700 perms here.